corehq/apps/accounting/management/commands/suppress_duplicate_invoices.py

```diff
@@ -0,0 +1,84 @@
+        parser.add_argument('month', type=int, help='The month of the statement period.')
+        parser.add_argument('note', type=str,
+                            help='A note to be added when crediting back the amount')
+
+    def handle(self, *args, **kwargs):
+        year = kwargs['year']
+        month = kwargs['month']
+        note = kwargs['note']
+
+        start_date, end_date = get_first_last_days(year, month)
+
+        # Filter the Invoice objects based on the date_start
+        invoices_of_given_date = Invoice.objects.filter(date_start__range=(start_date, end_date))
+
+        # Extract the subscription ids that have duplicate invoices
+        duplicate_subs_ids = invoices_of_given_date.values('subscription').annotate(
+            count=Count('id')).filter(count__gt=1).values_list('subscription', flat=True)
+
+        suppressed_count = 0
+        for sub_id in duplicate_subs_ids:
+            related_invoices = list(Invoice.objects.filter(
+                subscription_id=sub_id, date_start__range=(start_date, end_date)).order_by('-date_created'))
+            for invoice in related_invoices[:-1]:
+                if invoice.balance != invoice.subtotal:
+                    #credit them back
+                    self.revert_invoice_payment(invoice, note)
+                #suppress invoice
+                invoice.is_hidden_to_ops = True
+                print(f"Suppressing invoice {invoice.id} for domain {invoice.subscription.subscriber.domain} ",
+                      end='')
+                invoice.save()
+                suppressed_count += 1
+                print("✓")
+
+        self.stdout.write(self.style.SUCCESS(
+            'Successfully suppressed {} duplicate invoices for Statement Period: {}-{}'
+            .format(suppressed_count, year, month)))
+
+    def revert_invoice_payment(self, invoice, note):
+
+        print(f'Reverting payment for Invoice Id: {invoice.id}, Domain: {invoice.subscription.subscriber.domain}')
+        # Total Credit
+        payment_by_credit = -invoice.applied_credit
+
+        plan_subtotal, plan_deduction = get_subtotal_and_deduction(
+            invoice.lineitem_set.get_products().all()
+        )
+        if plan_deduction:
+            print(f'Adding plan credit: {plan_deduction}')
+            CreditLine.add_credit(amount=plan_deduction, subscription=invoice.subscription,
+                                  is_product=True, note=note)
+            payment_by_credit -= plan_deduction
+
+        for feature in FeatureType.CHOICES:
+            feature_subtotal, feature_deduction = get_subtotal_and_deduction(
+                invoice.lineitem_set.get_feature_by_type(feature).all()
+            )
+            if feature_deduction:
+                print(f"Adding {feature[0]} credit: {feature_deduction}")
+                CreditLine.add_credit(amount=feature_deduction, subscription=invoice.subscription,
```

Comment:
Known issue here: `feature[0]` would give me "User" and "SMS", however the `feature_type` in CreditLine model should be "USER" and "SMS"... I don't know how to do that, and I don't want to manually capitalize the word... Can I get some advice on this?

Reply:
> however the feature_type in CreditLine model should be "USER" and "SMS"

This was a bit of a surprise to me given that we have `FeatureType` class with static values. To verify this I queried production to see the values and I found out only `User` and `SMS` are used. 

Shared the queries on Slack.

I think you should be good with using `User` and `SMS` here. Let me know if this answers your questions or if I misunderstood it.

Reply:
Thank you! When I was playing around with CreditLine, I tried to generate each type of CreditLine in shell. So the “USER” record I saw in my local environment, is actually my own “wrong setup Credit Line”, not something we originally have in database. Will mark as resolved.